### PR TITLE
Add film

### PR DESCRIPTION
**Pull request by DimaFast**

### CODE REVIEW

File: src/api/films.js [hunk 1]
```diff
@@ -1,35 +1,50 @@
...
-  return index
+  const value = getItem(FILMS_LIST)
+  const newItem = { ...newData, id: filmId }
+
+  const allFilms = value.reduce(function (prev, curr) {
+    if (curr.id === filmId) {
+      curr = { ...curr, ...newItem }
+    }
+    return [...prev, curr]
+  }, [])
```
Comment (Di4ik3r) [inline src/api/films.js:45]: better use arrow function syntax:
```
const allFilms = value.reduce((prev, current) => { 
```

Comment (Di4ik3r) [reply src/api/films.js:45]: could be simplified, remove unnecessary if, use ternary

File: src/components/AddFilmModal/index.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      reset()
+    })
+  }
+
+  return (
+    <Modal show={open} centered>
+      <button
+        onClick={() => {
+          setOpen(false)
+        }}
```
Comment (Di4ik3r) [inline src/components/AddFilmModal/index.js:41]: remove unnecessary `{}`

File: src/components/AddFilmModal/index.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                    style: {
+                      position: 'absolute',
+                      marginTop: -30,
+                      transform: 'translateX(0)',
+                      left: '0',
+                    },
+                  },
+                }}
+              />
+            )}
```
Comment (Di4ik3r) [inline src/components/AddFilmModal/index.js:100]: to separate component

File: src/components/AddFilmModal/index.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                      position: 'absolute',
+                      top: -30,
+                      transform: 'translateX(0)',
+                      right: '0',
+                      left: 'auto',
+                    },
+                  },
+                }}
+              />
+            )}
```
Comment (Di4ik3r) [inline src/components/AddFilmModal/index.js:145]: to separate component, almost copypasta from above

File: src/components/AddFilmModal/index.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                    label: advertisingLabel[1] / 60,
+                    style: {
+                      position: 'absolute',
+                      top: -30,
+                      transform: 'translateX(0)',
+                      right: '0',
+                      left: 'auto',
+                    },
+                  },
+                }}
```
Comment (Di4ik3r) [inline src/components/AddFilmModal/index.js:143]: 90% copypaste

File: src/pages/changeList/index.js [hunk 6]
```diff
@@ -1,62 +1,165 @@
...
 
+  useEffect(() => {
+    getFilmsId(id).then((data) => {
+      setValue(data)
+      setFormValue(NAME_FIELD, data?.name)
+      setFormValue(TEXT_FIELD, data?.text)
+      setFormValue(TIME_TO_ADVERTISING, data.timeToAdvertising)
+      setFormValue(TIME_FILM, data?.timeFilm)
+      setAdvertisingLabel(data.timeToAdvertising)
+      setFilmTime(data?.timeFilm)
```
Comment (Di4ik3r) [inline src/pages/changeList/index.js:55]: why do you selectively choose to use `?.` or not to use?

File: src/pages/changeList/index.js [hunk 7]
```diff
@@ -1,62 +1,165 @@
...
+                    style: {
+                      position: 'absolute',
+                      marginTop: -30,
+                      transform: 'translateX(0)',
+                      left: '0',
+                    },
+                  },
+                }}
+              />
+            )}
```
Comment (Di4ik3r) [inline src/pages/changeList/index.js:115]: i saw this at least 2 times, move it to separate component

File: src/pages/changeList/index.js [hunk 8]
```diff
@@ -1,62 +1,165 @@
...
+                      position: 'absolute',
+                      top: -30,
+                      transform: 'translateX(0)',
+                      right: '0',
+                      left: 'auto',
+                    },
+                  },
+                }}
+              />
+            )}
```
Comment (Di4ik3r) [inline src/pages/changeList/index.js:159]: same as above

File: src/pages/list/index.js [hunk 9]
```diff
@@ -1,52 +1,56 @@
...
+    changeFilm(id, newData).then((data) => dispatch({ type: 'GET_FILMS', payload: data }))
 
-  const changeFilmId = (id, newData) => {
-    changeFilm(id, newData).then((index) => {
-      setValue([...index, newData])
-    })
-  }
+  useEffect(() => {
+    getFilms().then((data) => dispatch({ type: 'GET_FILMS', payload: data }))
+  }, [])
```
Comment (Di4ik3r) [inline src/pages/list/index.js:29]:
```
.then((data) => dispatch({ type: 'GET_FILMS', payload: data }))
```
duplicates 3 times

File: src/pages/signIn/index.js [hunk 10]
```diff
@@ -35,43 +34,45 @@ const SignIn = ({ setUserPro }) => {
...
           </label>
-          <input
-            {...register('email', { required: 'Please input email', defaultValue: '' })}
-            type="email"
-            className="inputSign"
+          <SimpleInput
+            type={EMAIL_FIELD}
+            register={{
+              ...register(EMAIL_FIELD, { required: 'Please input email', defaultValue: '' }),
+            }}
```
Comment (Di4ik3r) [inline src/pages/signIn/index.js:54]: you are creating new object and destructure in it `register` result
but you could call `register` without spreading
```
register={register(EMAIL_FIELD, { required: 'Please input email', defaultValue: '' })}
```

File: src/pages/signIn/index.js [hunk 11]
```diff
@@ -35,43 +34,45 @@ const SignIn = ({ setUserPro }) => {
...
-            type="password"
-            className="inputSign"
+          <label style={{ display: 'block', marginRight: 'auto' }} className="inputLabel">
+            {!errors?.password ? 'Input password' : 'Required password'}
+          </label>
+          <SimpleInput
+            type={PASSWORD_FIELD}
+            register={{
+              ...register(PASSWORD_FIELD, { required: 'Please input password', defaultValue: '' }),
+            }}
```
Comment (Di4ik3r) [inline src/pages/signIn/index.js:64]: same as above

File: src/pages/signUp/index.js [hunk 12]
```diff
@@ -14,42 +17,43 @@ const SignUp = () => {
...
-            type="email"
-            className="inputSign"
+          <SimpleInput
+            type={EMAIL_FIELD}
+            register={{
+              ...register(EMAIL_FIELD, {
+                required: 'Please input email',
+                defaultValue: '',
+              }),
+            }}
```
Comment (Di4ik3r) [inline src/pages/signUp/index.js:38]: without spreading

File: src/pages/signUp/index.js [hunk 13]
```diff
@@ -14,42 +17,43 @@ const SignUp = () => {
...
-            type="password"
-            className="inputSign"
+          <label style={{ display: 'block', marginRight: 'auto' }} className="inputLabel">
+            {!errors?.password ? 'Input Password' : 'Required Password'}
+          </label>
+          <SimpleInput
+            type={PASSWORD_FIELD}
+            register={{
+              ...register(PASSWORD_FIELD, { required: 'Please input password', defaultValue: '' }),
+            }}
```
Comment (Di4ik3r) [inline src/pages/signUp/index.js:47]: without spreading

**Review by Di4ik3r — CHANGES_REQUESTED** [review]

File: src/api/films.js (hunk 1 above)
Comment (DimaFast) [reply src/api/films.js:45]: ask you

**Pull request merged**